packages/compiler-core/src/utils.ts

```diff
@@ -430,6 +430,14 @@ export function toValidAssetId(
   name: string,
   type: 'component' | 'directive' | 'filter'
 ): string {
+  // see issue#4422, we need adding identifier on validAssetId if variable `name` has specific character
```

Comment:
I prefer this:

```javascript
return `_${type}_${name.replace(/[^\w]/g, function($1,$2){
	return $1==='-'?'_': name.charCodeAt($2)
})}`
```

Reply:
Yes, I think change it in this way is better.

Reply:
Done.